src/modeler/AutoRest.Swagger/Validation/ResourceModelValidation.cs

```diff
@@ -44,33 +47,19 @@ public class ResourceModelValidation: TypedRule<Dictionary<string, Schema>>
         /// </summary>
         /// <param name="definitions">Operation Definition to validate</param>
         /// <returns>true if the resource model is valid.false otherwise.</returns>
-        public override bool IsValid(Dictionary<string, Schema> definitions)
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Schema> entity, RuleContext context)
         {
-            foreach(string key in definitions.Keys)
+            var resModelList = entity.Where(pair => pair.Key.ToLower().Equals("resource"));
```

Comment:
@dsgouda do we need to enforce that this is the case? does it matter for this rule? or just the fact that the properties exist in the full set of properties of the model and that they are read-only is enough?